### PR TITLE
Fix setting timing flag in client creation

### PR DESCRIPTION
**Pull request by oporkka**
Fixed the timing flag logic so that `timing` given at the time of client
creation is used as fallback in case the `timing` is not given in
request parameters.

Added nock as a dependency to test the logic since it requires a real
`request` implementation to be called together with a client.

### CODE REVIEW

File: lib/client.ts [hunk 1]
```diff
@@ -578,6 +578,10 @@ export class ServiceClient {
 
     params.hostname = this.options.hostname;
     params.port = params.port || (params.protocol === "https:" ? 443 : 80);
+    params.timing =
+      params.timing !== undefined
+        ? params.timing
+        : this.options.timing || false;
```
Comment (ruiaraujo) [inline lib/client.ts]:
```suggestion
        : this.options.timing;
```

Comment (ruiaraujo) [reply lib/client.ts]: The options timing boolean should be normalized to be non null or undefined.

**Comment by shuhei** [comment]
👍

**Comment by ruiaraujo** [comment]
:+1:

**Comment by grassator** [comment]
:+1:

**Comment by shuhei** [comment]
:+1:

**Comment by grassator** [comment]
@shuhei Thanks a lot for the PRs. If you have an opportunity would be great to have travis CI setup with automatic NPM publish on version bump to make sure that features like this become immediately available to you and not blocked by manual publishing. If you have the time please provide your NPM username so I can add it to collaborators. Preferably send it through internal channels.

**Comment by shuhei** [comment]
@grassator Sorry for this late reply. I created an issue for automatic publish. https://github.com/zalando-incubator/perron/issues/99

I'll contact you in an internal channel after holidays.

**Pull request merged**